File: 009_PalindromeNumber.py

#60.64%
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False
        if x < 10:
            return True
        
        origX = x
        rstX = 0
        while x:
            rstX = rstX * 10 + x % 10
            x = x / 10
        
        if origX == rstX:
            return True
        return False

File: 009_PalindromeNumber.cpp

class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0)
			return false;
		if(x <= 9)
			return true;
			
		int origX = x;
		int rvsX = 0;
		while(x)
		{
			rvsX = rvsX * 10 + x % 10;
			x /= 10;
		}

		if(rvsX == origX)
			return true;
		return false;
    }
};

//75.55%
class Solution1 {
public:
    bool isPalindrome(int x) {
        if(x < 0)
			return false;
		if(x >= 0 && x <= 9)
			return true;
			
		int origX = x;
		int rvsX = 0;
		while(x)
		{
			rvsX = rvsX * 10 + x % 10;
			x /= 10;
		}
		
		if(rvsX >= INT_MAX)
			return false;
		if(rvsX == origX)
			return true;
		return false;
    }
};